substrate/frame/identity/src/types.rs

```diff
@@ -234,24 +231,25 @@ impl<Balance: Encode + Decode + MaxEncodedLen + Copy + Clone + Debug + Eq + Part
 pub trait IdentityInformationProvider:
 	Encode + Decode + MaxEncodedLen + Clone + Debug + Eq + PartialEq + TypeInfo
 {
-	/// Type capable of representing all of the fields present in the identity information as bit
-	/// flags in `u64` format.
-	type IdentityField: Clone + Debug + Eq + PartialEq + TypeInfo + U64BitFlag;
+	/// Type capable of representing all of the fields present in the identity information.
+	type FieldsBitFlags: Encode
```

Comment:
`BitFlags` had the advantage of validating the input automatically and now we lost that. I was thinking we could add a function in this trait `fn is_valid(fields: Self::FieldsBitFlags) -> bool` to validate input so we don't end up with people setting fields that don't exist.